### PR TITLE
Update CML to regenerated version and use instead of Pallas 

### PR DESCRIPTION
**Pull request by gostkin**
This replaces pallas in parsing completely

This PR was tested with preprod and preview networks

### CODE REVIEW

File: indexer/tasks/src/byron/byron_block.rs [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 use crate::dsl::task_macro::*;
 use crate::era_common::block_from_hash;
+use cml_core::serialization::ToBytes;
+use cml_crypto::{blake2b256, RawBytesEncoding};
+use cml_multi_era::byron::block::ByronBlock;
+use cml_multi_era::MultiEraBlock;
 use entity::{block::EraValue, sea_orm::Set};
 use hex::ToHex;
-use pallas::ledger::primitives::{byron, Fragment};
+use sea_orm::sea_query::all;
```
Comment (SebastienGllmt) [inline indexer/tasks/src/byron/byron_block.rs:10]: why was this line added?

File: indexer/tasks/src/multiera/dex/minswap_v1.rs [hunk 2]
```diff
@@ -24,18 +25,23 @@ impl Dex for MinSwapV1 {
...
         if let Some((output, datum)) = filter_outputs_and_datums_by_hash(
             &tx.outputs(),
             &[POOL_SCRIPT_HASH1, POOL_SCRIPT_HASH2],
-            &tx.plutus_data(),
+            &tx_witness.plutus_datums.clone().unwrap_or_default(),
         )
         .first()
         {
-            let datum = datum.to_json();
+            let pallas_datum = pallas::ledger::primitives::alonzo::PlutusData::decode_fragment(
```
Comment (SebastienGllmt) [inline indexer/tasks/src/multiera/dex/minswap_v1.rs:40]: replace this with the JSON conversion in CML. You may need https://github.com/dcSpark/cardano-multiplatform-lib/pull/300

File: indexer/tasks/src/multiera/multiera_address.rs [hunk 3]
```diff
@@ -131,15 +132,15 @@ fn queue_certificate(
...
     // TODO: what's the policy for handling options? At the moment of writing, all certificates
     // are "alonzo-compatible", but that might change in a future HF. Should Carp skip data that
     // it doesn't understand or instead panic? For now, opting to panic as it seems to be what's
     // used for other options.
-    match cert.as_alonzo().unwrap() {
-        Certificate::StakeDelegation(credential, pool) => {
-            let credential = credential.encode_fragment().unwrap();
+    match cert {
+        MultiEraCertificate::StakeDelegation(delegation) => {
+            let credential = delegation.stake_credential.to_canonical_cbor_bytes();
```
Comment (SebastienGllmt) [inline indexer/tasks/src/multiera/multiera_address.rs:143]: note that `to_cannonical` isn't the same representation of what is onchain. I don't think that's necessarily a bad thing for fields that aren't critical (block and transactions), but if we decide to use canonical instead of the real encoding onchain we need to specify that somewhere

This comment applies not just here, but other places as well

File: indexer/Cargo.toml [hunk 4]
```diff
@@ -21,13 +21,15 @@ tasks = { path = "tasks" }
 # [indexer]
 anyhow = { version = "1.0.69" }
 async-trait = { version = "0.1.64" }
-cardano-multiplatform-lib = { git = "https://github.com/dcSpark/cardano-multiplatform-lib", branch = "metadata-and-addr" }
+cml-chain = { git = "https://github.com/dcSpark/cardano-multiplatform-lib", rev = "c5740393f10b7a68aa4719fbbef5a49170501ef6" }
+cml-core = { git = "https://github.com/dcSpark/cardano-multiplatform-lib", rev = "c5740393f10b7a68aa4719fbbef5a49170501ef6" }
+cml-crypto = { git = "https://github.com/dcSpark/cardano-multiplatform-lib", rev = "c5740393f10b7a68aa4719fbbef5a49170501ef6" }
+cml-multi-era = { git = "https://github.com/dcSpark/cardano-multiplatform-lib", rev = "c5740393f10b7a68aa4719fbbef5a49170501ef6" }
```
[the remark this answers is not in the document]
Comment (gostkin) [reply indexer/Cargo.toml:27]: will we do a CML release btw?

File: indexer/src/sinks/cardano.rs [hunk 5]
```diff
@@ -254,14 +255,14 @@ impl StoppableService for CardanoSink {
...
-        pallas::ledger::traverse::Era::Allegra => EraValue::Allegra,
-        pallas::ledger::traverse::Era::Mary => EraValue::Mary,
-        pallas::ledger::traverse::Era::Alonzo => EraValue::Alonzo,
-        pallas::ledger::traverse::Era::Babbage => EraValue::Babbage,
+        MultiEraBlock::Byron(_) => EraValue::Byron,
+        MultiEraBlock::Shelley(_) => EraValue::Shelley,
+        MultiEraBlock::Allegra(_) => EraValue::Allegra,
+        MultiEraBlock::Mary(_) => EraValue::Mary,
+        MultiEraBlock::Alonzo(_) => EraValue::Alonzo,
+        MultiEraBlock::Babbage(_) => EraValue::Babbage,
```
[the remark this answers is not in the document]
Comment (gostkin) [reply indexer/src/sinks/cardano.rs:265]: yes

File: indexer/tasks/src/byron/byron_block.rs (hunk 1 above)
Comment (gostkin) [reply indexer/tasks/src/byron/byron_block.rs:10]: typo

File: indexer/tasks/src/multiera/multiera_address.rs (hunk 3 above)
Comment (gostkin) [reply indexer/tasks/src/multiera/multiera_address.rs:143]: we can switch to `.to_cbor_bytes()` for safety

Comment (SebastienGllmt) [reply indexer/tasks/src/multiera/multiera_address.rs:143]: Yeah I would rather we use the same encoding as what is onchain whenever possible instead of converting it to cannonical cbor

Comment (gostkin) [reply indexer/tasks/src/multiera/multiera_address.rs]: ok fixed

File: indexer/tasks/src/multiera/dex/minswap_v1.rs (hunk 2 above)
Comment (gostkin) [reply indexer/tasks/src/multiera/dex/minswap_v1.rs:40]: how do I import that? I can't see a feature / method similar to `to_json`

Comment (gostkin) [reply indexer/tasks/src/multiera/dex/minswap_v1.rs]: fixed
